src/test/rgw/bucket_notification/test_bn.py

```diff
@@ -1730,6 +1730,50 @@ def test_ps_s3_creation_triggers_on_master_ssl():
         del os.environ['RABBITMQ_CONFIG_FILE']
 
 
+def _get_body(response):
+    body = response['Body']
+    got = body.read()
+    if type(got) is bytes:
+        got = got.decode()
+    return got
+
+
+@attr('sample_test')
+def test_http_post_object_upload():
+    """ test that uploads object using HTTP POST """
+
+    import boto3
+    from collections import OrderedDict
+    import requests
+
+    endpoint = "http://%s:%d" % (get_config_host(),get_config_port())
+
+    conn1 = boto3.client(service_name='s3',
+                         aws_access_key_id=get_access_key(),
+                         aws_secret_access_key=get_secret_key(),
+                         endpoint_url=endpoint
+                        )
+
+    bucket_name = gen_bucket_name()
+
+    key_name = 'foo.txt'
+
+    resp = conn1.generate_presigned_post(Bucket=bucket_name, Key=key_name,)
+
+    url = resp['url']
+
+    payload = OrderedDict([("key" , "foo.txt"),("acl" , "public-read"),\
+    ("Content-Type" , "text/plain"),('file', ('bar'))])
+
+    bucket = conn1.create_bucket(ACL='public-read-write', Bucket=bucket_name)
+
+    r = requests.post(url, files=payload, verify=True)
+    assert_equal(r.status_code, 204)
+    response = conn1.get_object(Bucket=bucket_name, Key='foo.txt')
```

Comment:
no need to do the "get_object" part
but you should add notifications definitions for POST messages, and verify them